zigpy/topology.py

```diff
@@ -166,6 +174,7 @@ async def _scan(
             LOGGER.debug(
                 "Scanning topology (%d/%d) of %s", index + 1, len(devices), device
             )
+            self.scan_progress = index / len(devices)
```

Comment:
If you are subscribing to scan progress, it would be better to emit an event (`self.listener_event("scan_progress, index, len(devices))`) instead of computing the percentage and storing it as an attribute. This should also make it easier to test.